user-interface/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -447,12 +449,21 @@ private void addExperimentalGroups(
     ExperimentId experimentId = context.experimentId().orElseThrow();
     Result<Collection<ExperimentalGroup>, ResponseCode> result = experimentInformationService.addExperimentalGroupsToExperiment(
         experimentId, experimentalGroupDTOS);
-    result.onError(error -> {
-      throw new ApplicationException(
-          "Could not save one or more experimental groups %s %nReason: %s".formatted(
-              Arrays.toString(
-                  experimentalGroupContents.toArray()), error));
-    });
+    if(result.isError()) {
+      if (result.getError().equals(ResponseCode.CONDITION_EXISTS)) {
+        throw new ApplicationException(ErrorCode.DUPLICATE_GROUP_SELECTED,
+            ErrorParameters.empty());
+      }
```

Comment:
Nice! However without specifying a message for this exception the UIExceptionhandler will not know why the exception occured leading to the following unspecific email:

<img width="569" alt="Bildschirmfoto 2023-11-08 um 11 00 51" src="https://github.com/qbicsoftware/data-manager-app/assets/29627977/a6a7c20d-e853-4ee7-98b0-e401a92a7295">

and unspecific log message

![Bildschirmfoto 2023-11-08 um 10 35 29](https://github.com/qbicsoftware/data-manager-app/assets/29627977/c14919e9-7a3f-4c8e-bd29-b196f0ced4a5)

Reply:
This can be fixed easily by providing a message within the ApplicationException 

```suggestion
        throw new ApplicationException("Duplicate experimental group was selected", ErrorCode.DUPLICATE_GROUP_SELECTED,
            ErrorParameters.empty());
      }
```